File: sort_py/merge_sort.py
arr = [21, 75, 40, 89, 72, 2, 45, 32, 28, 7, 56, 58, 60, 64, 50]


def merge(la, lb):
    tmp = []
    while la or lb:
        if not la:
            tmp.extend(lb)
            return tmp
        elif not lb:
            tmp.extend(la)
            return tmp
        if la[0] < lb[0]:
            tmp.append(la.pop(0))
        else:
            tmp.append(lb.pop(0))
    return tmp


def divide(a):
    if len(a) > 1:
        mid = len(a) // 2
        return merge(divide(a[0: mid]), divide(a[mid: len(a)]))
    else:
        return a


if __name__ == '__main__':
    res = divide(arr)
    print(res)

File: sort_js/merge_sort.js
var arr = [21, 75, 40, 89, 72, 2, 45, 32, 28, 7, 56, 58, 60, 64, 50];


function merge(la, lb) {
    var tmp = [];
    while (la.length > 0 || lb.length > 0) {
        if (la.length === 0) {
            tmp = tmp.concat(lb);
            return tmp;
        }
        else if (lb.length === 0) {
            tmp = tmp.concat(la);
            return tmp;
        }
        la[0] > lb[0] ? tmp.push(lb.shift()) : tmp.push(la.shift());
    }
    return tmp;
}

function divide(a) {
    if (a.length > 1) {
        var mid = parseInt(a.length / 2);
        return merge(divide(a.slice(0, mid)), divide(a.slice(mid, a.length)));
    }
    else {
        return a;
    }
}

(function main() {
    var res = divide(arr);
    console.log(res);
})();